src/snowflake/snowpark/mock/_stored_procedure.py

```diff
@@ -90,13 +75,11 @@ def __call__(
 
                 # If expression does not define its datatype we cannot verify it's compatibale.
                 # This is potentially unsafe.
-                if expr.datatype and not sproc_types_are_compatible(
+                if expr.datatype and not types_are_compatible(
                     expr.datatype, expected_type
                 ):
-                    SnowparkLocalTestingException.raise_from_error(
```

Comment:
Why getting rid of the original error?

Reply:
The old call looked like this:
```
SnowparkLocalTestingException.raise_from_error(
    ValueError(
        f"Unexpected type {expr.datatype} for sproc argument of type {expected_type}"
    )
)
```
In all the other cases I've seen `raise_from_error` it has provided additional useful information because the error is raised in one spot and then re-raised wherever we have `raise_from_error`. This allows preserving the traceback from the original error while still having our own error. Here there is no additional traceback or useful information by chaining the error in my opinion.